### PR TITLE
fix: correctly set pk/indexes when table name changes

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

* correctly set pk/indexes when table name changes
* add test for schema query drift
* Add DISTINCT to pg and mysql schema queries (reduces the size of the schema csv)
* recreate all test schemas

##### CDK / CloudFormation Parameters Changed

N/A

#### Issue #, if available

N/A

#### Description of how you validated changes

Unit testing

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-schema-generator/src/__tests__/__snapshots__/generate.test.ts.snap [hunk 1]
```diff
@@ -80,6 +113,38 @@ type Story @refersTo(name: \\"stories\\") @model {
...
 
+exports[`generate graphqlSchemaFromSQLSchema creates graphql schema from "postgres" "newsNameChange" schema 1`] = `
+"input AMPLIFY {
+ engine: String = \\"postgres\\"  
+}
+
+
+type Article @refersTo(name: \\"articles\\") @model {
+  pubId: Int! @refersTo(name: \\"pub_id\\") @primaryKey(sortKeyFields: [\\"pubType\\"])
+  pubType: String! @refersTo(name: \\"pub_type\\") @default(value: \\"'A'::bpchar\\")
```
Comment (phani-srikar) [inline packages/amplify-graphql-schema-generator/src/__tests__/__snapshots__/generate.test.ts.snap:124]: The default values generated here don't look right. We would need to remove the `::<type>` part.

File: packages/amplify-graphql-schema-generator/src/datasource-adapter/pg-string-datasource-adapter.ts [hunk 2]
```diff
@@ -118,7 +118,9 @@ export class PostgresStringDataSourceAdapter extends StringDataSourceAdapter {
   }
 
   public getPrimaryKey(tableName: string): Index | null {
-    const key = this.indexes.find((index) => index.tableName === tableName && index.indexName === `${tableName}_pkey`);
+    const key = this.indexes.find(
+      (index) => index.tableName === tableName && index.indexName.match(/.*_pkey/) && index.columns.includes(index.columnName),
```
Comment (phani-srikar) [inline packages/amplify-graphql-schema-generator/src/datasource-adapter/pg-string-datasource-adapter.ts:122]: This is still using name matching on index to determine PK. If we have indices named as `my_index_pk` we would would mistake them to be primaryKey constraints?

File: packages/amplify-graphql-schema-generator/src/__tests__/__snapshots__/generate.test.ts.snap (hunk 1 above)
Comment (dpilch) [reply packages/amplify-graphql-schema-generator/src/__tests__/__snapshots__/generate.test.ts.snap:124]: Looks like this an existing bug. I'll fix this in a separate PR.

File: packages/amplify-graphql-schema-generator/src/datasource-adapter/pg-string-datasource-adapter.ts (hunk 2 above)
Comment (dpilch) [reply packages/amplify-graphql-schema-generator/src/datasource-adapter/pg-string-datasource-adapter.ts:122]: I believe Postgres would throw an error here when trying to create the table. In our `news` test schema if I change `CONSTRAINT publications_superkey UNIQUE (pub_id, pub_type)` to `CONSTRAINT publications_pkey UNIQUE (pub_id, pub_type)` the query fails with: `ERROR:  relation "publications_pkey" already exists`.

**Review by phani-srikar — APPROVED** [review]

**Review by palpatim — APPROVED** [review]

**Pull request merged**